etc/kayobe/ansible/ubuntu-upgrade.yml

```diff
@@ -0,0 +1,103 @@
+---
+# To prevent Ansible role dependency errors, this playbook requires that environment variable
+# ANSIBLE_ROLES_PATH is defined and includes '$KAYOBE_PATH/ansible/roles' at seed-hypervisor.
+- name: Migrate hosts from Ubuntu Focal 20.04 to Jammy 22.04
+  hosts: overcloud:infra-vms:seed
+  vars:
+    ansible_python_interpreter: /usr/bin/python3
+  tasks:
+    - name: Assert that hosts are running Ubuntu Focal
+      assert:
+        that:
+          - ansible_facts.distribution == 'Ubuntu'
+          - ansible_facts.distribution_major_version == '20'
+          - ansible_facts.distribution_release == 'focal'
+          - os_distribution == 'ubuntu'
+        fail_msg: >-
+          This playbook is only designed for Ubuntu Focal 20.04 hosts. Ensure
+          that you are limiting it to only run on Focal hosts and
+          os_distribution is set to ubuntu.
+
+    - name: Ensure apt packages are up to date
+      apt:
+        update_cache: true
+        upgrade: yes
+      become: true
+
+    # NOTE: We cannot use apt_repository here because definitions must exist within the standard repos.list
+    - name: Ensure Jammy repo definitions exist in sources.list
+      blockinfile:
+        path: /etc/apt/sources.list
+        block: |
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-updates main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_url }} jammy-backports main restricted universe multiverse
+          deb {{ stackhpc_repo_ubuntu_jammy_security_url }} jammy-security main restricted universe multiverse
+      become: true
+
+    - name: Ensure do-release-upgrade is installed
+      package:
+        name: ubuntu-release-upgrader-core
+        state: latest
+      become: true
+
+    - name: Do release upgrade
+      command: do-release-upgrade -f DistUpgradeViewNonInteractive
+      become: true
+
+    - name: Ensure Jammy repo definitions do not exist in sources.list
+      blockinfile:
+        path: /etc/apt/sources.list
+        state: absent
```

Comment:
FWIW, I think you can omit the block when setting state=absent, and avoid keeping it in sync. I don't know what it would do if it did not match the existing block. You can pass the `marker` argument to uniquely identify the block.